### PR TITLE
Show 'positive' result even if just anti_s is positive

### PR DESCRIPTION
**Pull request by juliatan**
# Description

Notion: https://www.notion.so/joinzoe/Antibody-Results-Swap-Bug-12dceff02dac4745bd3578444c1b37f2

- Previously, a 'positive' result was only shown in the list view, if `anti_n` was positive. 
- This PR also shows a 'positive' result if `anti_s` is positive.

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

## Screenshots

[comment]: <> (Please attach screenshots showing the change, if relevant)

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Checklist (for merger)

- [ ] Make sure the right merge strategy is chosen, commit merge into master and squash merge into develop.

### CODE REVIEW

File: src/features/covid-tests/components/CovidTestRow.tsx [hunk 1]
```diff
@@ -56,10 +64,12 @@ export const CovidTestRow: React.FC<TProps> = ({ item, testID }) => {
...
       testID={testID}
     >
-      <Image source={getRowIcon(testResult)} style={styles.tick} />
-      <RegularText style={[testResult === 'waiting' && styles.pendingText]}>{formatTestDate(item)}</RegularText>
+      <Image source={getRowIcon(testResult())} style={styles.tick} />
+      <RegularText style={[testResult() === 'waiting' && styles.pendingText]}>{formatTestDate(item)}</RegularText>
       <View style={{ flex: 1 }} />
-      <RegularText style={[testResult === 'waiting' && styles.pendingText]}>{formatTestResult(testResult)}</RegularText>
+      <RegularText style={[testResult() === 'waiting' && styles.pendingText]}>
+        {formatTestResult(testResult())}
```
Comment (aikewoody) [inline src/features/covid-tests/components/CovidTestRow.tsx:71]: It's an inefficient coding pattern to call the `testResult` 4 times each render when you know the result will be the same. I suggest to call the function just once, for exampe:

```
const testResult = (() => {
...
})()
```

or

```
let testResult = 'waiting';
if (item.result) {
  testResult = item.result;
} else if (item.anti_n) {
  testResult = item.anti_n === 'positive' || item.anti_s === 'positive' ? 'positive' : item.anti_n;
}
```

Comment (aikewoody) [reply src/features/covid-tests/components/CovidTestRow.tsx:71]: Or maybe even better wrap the function call in an `React.useMemo`

Comment (juliatan) [reply src/features/covid-tests/components/CovidTestRow.tsx]: Yup, great point! Thanks - just fixed with `React.useMemo`.

File: src/features/covid-tests/components/CovidTestRow.tsx [hunk 2]
```diff
@@ -15,7 +15,15 @@ type TProps = {
...
-  const testResult = item.result ? item.result : item.anti_n ? item.anti_n : 'waiting';
+  const testResult = React.useMemo(() => {
+    if (item.result) {
+      return item.result;
+    }
+    if (item.anti_n) {
+      return item.anti_n === 'positive' || item.anti_s === 'positive' ? 'positive' : item.anti_n;
+    }
+    return 'waiting';
+  }, [item.result, item.anti_n]);
```
Comment (aikewoody) [inline src/features/covid-tests/components/CovidTestRow.tsx:26]: It also depends on `item.anti_s`

**Review by aikewoody — APPROVED** [review]

**Pull request merged**